libsrc/telestrat/gotoxy.s

```diff
@@ -19,8 +18,43 @@ gotoxy: jsr     popa            ; Get Y
 ; This function moves only the display cursor; it does not move the prompt position.
 ; In telemon, there is a position for the prompt, and another for the cursor.
 
-        sta     SCRY
-        jsr     popa
-        sta     SCRX
-        rts
+    sta     SCRY
+    
+
+    jsr     popa
+    sta     SCRX
+
+    jmp     update_adscr          ; Update adress video ram position when SCRY is modified 
+    
+.endproc
+
+.proc update_adscr
+; Force to set again color if cursor moves       
+; $FF is used because we know that it's impossible to have this value with a color
+; It prevents a bug : If bgcolor or textcolor is set to black for example with no char displays,
+; next cputsxy will not set the attribute if y coordinate changes
+    lda     #$FF                  
+    sta     OLD_CHARCOLOR         
+    sta     OLD_BGCOLOR           
+
+    lda     #<SCREEN
+    sta     ADSCRL
+
+    lda     #>SCREEN
+    sta     ADSCRH
+
+    ldy     SCRY
+    beq     out
+loop:
+    lda     ADSCRL          
+    clc
+    adc     #$28
```

Comment:
You should change `$28` to `SCREEN_XSIZE`.

Reply:
Fixed